<!DOCTYPE html><html lang="zh-CN" data-theme="light"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0,viewport-fit=cover"><title>GSSG | Evanescence's Blog</title><meta name="author" content="Henry"><meta name="copyright" content="Henry"><meta name="format-detection" content="telephone=no"><meta name="theme-color" content="ffffff"><meta name="description" content="有源码kkdey&#x2F;GSSG: Gene Set + S2G strategy annotations analyzed for disease architecture 简介GSSG : Combine SNP-gene linking strategy with Gene Sets A pipeline to perform the following steps.  Generate">
<meta property="og:type" content="article">
<meta property="og:title" content="GSSG">
<meta property="og:url" content="http://evanescence0515.github.io/2026/01/12/GSSG/index.html">
<meta property="og:site_name" content="Evanescence&#39;s Blog">
<meta property="og:description" content="有源码kkdey&#x2F;GSSG: Gene Set + S2G strategy annotations analyzed for disease architecture 简介GSSG : Combine SNP-gene linking strategy with Gene Sets A pipeline to perform the following steps.  Generate">
<meta property="og:locale" content="zh_CN">
<meta property="og:image" content="http://evanescence0515.github.io/img/epidemic_pic.png">
<meta property="article:published_time" content="2026-01-12T09:33:57.000Z">
<meta property="article:modified_time" content="2026-01-12T09:33:57.000Z">
<meta property="article:author" content="Henry">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="http://evanescence0515.github.io/img/epidemic_pic.png"><script type="application/ld+json">{
  "@context": "https://schema.org",
  "@type": "BlogPosting",
  "headline": "GSSG",
  "url": "http://evanescence0515.github.io/2026/01/12/GSSG/",
  "image": "http://evanescence0515.github.io/img/epidemic_pic.png",
  "datePublished": "2026-01-12T09:33:57.000Z",
  "dateModified": "2026-01-12T09:33:57.000Z",
  "author": [
    {
      "@type": "Person",
      "name": "Henry",
      "url": "http://evanescence0515.github.io"
    }
  ]
}</script><link rel="shortcut icon" href="/img/favicon.png"><link rel="canonical" href="http://evanescence0515.github.io/2026/01/12/GSSG/index.html"><link rel="preconnect" href="//cdn.jsdelivr.net"/><link rel="preconnect" href="//busuanzi.ibruce.info"/><link rel="stylesheet" href="/css/index.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free/css/all.min.css"><script>
    (() => {
      
    const saveToLocal = {
      set: (key, value, ttl) => {
        if (!ttl) return
        const expiry = Date.now() + ttl * 86400000
        localStorage.setItem(key, JSON.stringify({ value, expiry }))
      },
      get: key => {
        const itemStr = localStorage.getItem(key)
        if (!itemStr) return undefined
        const { value, expiry } = JSON.parse(itemStr)
        if (Date.now() > expiry) {
          localStorage.removeItem(key)
          return undefined
        }
        return value
      }
    }

    window.btf = {
      saveToLocal,
      getScript: (url, attr = {}) => new Promise((resolve, reject) => {
        const script = document.createElement('script')
        script.src = url
        script.async = true
        Object.entries(attr).forEach(([key, val]) => script.setAttribute(key, val))
        script.onload = script.onreadystatechange = () => {
          if (!script.readyState || /loaded|complete/.test(script.readyState)) resolve()
        }
        script.onerror = reject
        document.head.appendChild(script)
      }),
      getCSS: (url, id) => new Promise((resolve, reject) => {
        const link = document.createElement('link')
        link.rel = 'stylesheet'
        link.href = url
        if (id) link.id = id
        link.onload = link.onreadystatechange = () => {
          if (!link.readyState || /loaded|complete/.test(link.readyState)) resolve()
        }
        link.onerror = reject
        document.head.appendChild(link)
      }),
      addGlobalFn: (key, fn, name = false, parent = window) => {
        if (!false && key.startsWith('pjax')) return
        const globalFn = parent.globalFn || {}
        globalFn[key] = globalFn[key] || {}
        globalFn[key][name || Object.keys(globalFn[key]).length] = fn
        parent.globalFn = globalFn
      }
    }
  
      
      const activateDarkMode = () => {
        document.documentElement.setAttribute('data-theme', 'dark')
        if (document.querySelector('meta[name="theme-color"]') !== null) {
          document.querySelector('meta[name="theme-color"]').setAttribute('content', '#0d0d0d')
        }
      }
      const activateLightMode = () => {
        document.documentElement.setAttribute('data-theme', 'light')
        if (document.querySelector('meta[name="theme-color"]') !== null) {
          document.querySelector('meta[name="theme-color"]').setAttribute('content', 'ffffff')
        }
      }

      btf.activateDarkMode = activateDarkMode
      btf.activateLightMode = activateLightMode

      const theme = saveToLocal.get('theme')
    
          theme === 'dark' ? activateDarkMode() : theme === 'light' ? activateLightMode() : null
        
      
      const asideStatus = saveToLocal.get('aside-status')
      if (asideStatus !== undefined) {
        document.documentElement.classList.toggle('hide-aside', asideStatus === 'hide')
      }
    
      
    const detectApple = () => {
      if (/iPad|iPhone|iPod|Macintosh/.test(navigator.userAgent)) {
        document.documentElement.classList.add('apple')
      }
    }
    detectApple()
  
    })()
  </script><script>const GLOBAL_CONFIG = {
  root: '/',
  algolia: undefined,
  localSearch: {"path":"/search.xml","preload":false,"top_n_per_article":1,"unescape":false,"languages":{"hits_empty":"未找到符合您查询的内容：${query}","hits_stats":"共找到 ${hits} 篇文章"}},
  translate: undefined,
  highlight: {"plugin":"highlight.js","highlightCopy":true,"highlightLang":true,"highlightHeightLimit":false,"highlightFullpage":false,"highlightMacStyle":false},
  copy: {
    success: '复制成功',
    error: '复制失败',
    noSupport: '浏览器不支持'
  },
  relativeDate: {
    homepage: false,
    post: false
  },
  runtime: '',
  dateSuffix: {
    just: '刚刚',
    min: '分钟前',
    hour: '小时前',
    day: '天前',
    month: '个月前'
  },
  copyright: undefined,
  lightbox: 'null',
  Snackbar: undefined,
  infinitegrid: {
    js: 'https://cdn.jsdelivr.net/npm/@egjs/infinitegrid/dist/infinitegrid.min.js',
    buttonText: '加载更多'
  },
  isPhotoFigcaption: false,
  islazyloadPlugin: false,
  isAnchor: false,
  percent: {
    toc: true,
    rightside: false,
  },
  autoDarkmode: false
}</script><script id="config-diff">var GLOBAL_CONFIG_SITE = {
  title: 'GSSG',
  isHighlightShrink: false,
  isToc: true,
  pageType: 'post'
}</script><meta name="generator" content="Hexo 7.3.0"></head><body><div id="web_bg" style="background-color: #EEEEEE;"></div><div id="sidebar"><div id="menu-mask"></div><div id="sidebar-menus"><div class="avatar-img text-center"><img src="/img/my_icon.jpg" onerror="this.onerror=null;this.src='/img/friend_404.gif'" alt="avatar"/></div><div class="site-data text-center"><a href="/archives/"><div class="headline">文章</div><div class="length-num">35</div></a><a href="/tags/"><div class="headline">标签</div><div class="length-num">9</div></a><a href="/categories/"><div class="headline">分类</div><div class="length-num">3</div></a></div><div class="menus_items"><div class="menus_item"><a class="site-page" href="/"><i class="fa-fw fas fa-home"></i><span> Home</span></a></div><div class="menus_item"><a class="site-page" href="/archives/"><i class="fa-fw fas fa-archive"></i><span> Archives</span></a></div><div class="menus_item"><a class="site-page" href="/tags/"><i class="fa-fw fas fa-tags"></i><span> Tags</span></a></div><div class="menus_item"><a class="site-page" href="/categories/"><i class="fa-fw fas fa-folder-open"></i><span> Categories</span></a></div><div class="menus_item"><span class="site-page group"><i class="fa-fw fas fa-list"></i><span> List</span><i class="fas fa-chevron-down"></i></span><ul class="menus_item_child"><li><a class="site-page child" href="/tools/"><i class="fa-fw fas fa-tools"></i><span> Tools</span></a></li></ul></div><div class="menus_item"><a class="site-page" href="/link/"><i class="fa-fw fas fa-link"></i><span> Link</span></a></div><div class="menus_item"><a class="site-page" href="/about/"><i class="fa-fw fas fa-heart"></i><span> About</span></a></div></div></div></div><div class="post" id="body-wrap"><header class="post-bg" id="page-header" style="background-image: url(/img/epidemic_pic.png);"><nav id="nav"><span id="blog-info"><a class="nav-site-title" href="/"><span class="site-name">Evanescence's Blog</span></a><a class="nav-page-title" href="/"><span class="site-name">GSSG</span><span class="site-name"><i class="fa-solid fa-circle-arrow-left"></i><span>  返回首页</span></span></a></span><div id="menus"><div id="search-button"><span class="site-page social-icon search"><i class="fas fa-search fa-fw"></i><span> 搜索</span></span></div><div class="menus_items"><div class="menus_item"><a class="site-page" href="/"><i class="fa-fw fas fa-home"></i><span> Home</span></a></div><div class="menus_item"><a class="site-page" href="/archives/"><i class="fa-fw fas fa-archive"></i><span> Archives</span></a></div><div class="menus_item"><a class="site-page" href="/tags/"><i class="fa-fw fas fa-tags"></i><span> Tags</span></a></div><div class="menus_item"><a class="site-page" href="/categories/"><i class="fa-fw fas fa-folder-open"></i><span> Categories</span></a></div><div class="menus_item"><span class="site-page group"><i class="fa-fw fas fa-list"></i><span> List</span><i class="fas fa-chevron-down"></i></span><ul class="menus_item_child"><li><a class="site-page child" href="/tools/"><i class="fa-fw fas fa-tools"></i><span> Tools</span></a></li></ul></div><div class="menus_item"><a class="site-page" href="/link/"><i class="fa-fw fas fa-link"></i><span> Link</span></a></div><div class="menus_item"><a class="site-page" href="/about/"><i class="fa-fw fas fa-heart"></i><span> About</span></a></div></div><div id="toggle-menu"><span class="site-page"><i class="fas fa-bars fa-fw"></i></span></div></div></nav><div id="post-info"><h1 class="post-title">GSSG</h1><div id="post-meta"><div class="meta-firstline"><span class="post-meta-date"><i class="far fa-calendar-alt fa-fw post-meta-icon"></i><span class="post-meta-label">发表于</span><time class="post-meta-date-created" datetime="2026-01-12T09:33:57.000Z" title="发表于 2026-01-12 17:33:57">2026-01-12</time><span class="post-meta-separator">|</span><i class="fas fa-history fa-fw post-meta-icon"></i><span class="post-meta-label">更新于</span><time class="post-meta-date-updated" datetime="2026-01-12T09:33:57.000Z" title="更新于 2026-01-12 17:33:57">2026-01-12</time></span></div><div class="meta-secondline"><span class="post-meta-separator">|</span><span class="post-meta-pv-cv" id="" data-flag-title=""><i class="far fa-eye fa-fw post-meta-icon"></i><span class="post-meta-label">浏览量:</span><span id="busuanzi_value_page_pv"><i class="fa-solid fa-spinner fa-spin"></i></span></span></div></div></div></header><main class="layout" id="content-inner"><div id="post"><article class="container post-content" id="article-container"><p>有源码<a target="_blank" rel="noopener" href="https://github.com/kkdey/GSSG/tree/master">kkdey&#x2F;GSSG: Gene Set + S2G strategy annotations analyzed for disease architecture</a></p>
<h1 id="简介"><a href="#简介" class="headerlink" title="简介"></a>简介</h1><p>GSSG : Combine SNP-gene linking strategy with Gene Sets A pipeline to perform the following steps.</p>
<ul>
<li>Generate gene programs for analysis</li>
<li>Link SNPs to genes in a gene program to generate a genome-wide annotation</li>
<li>Perform disease heritability enrichment analysis using S-LDSC<br>这里我们将主要关注其第二篇文章的工作：将GWAS数据与scRNA结合以识别细胞种类和细胞过程(<a target="_blank" rel="noopener" href="https://www.biorxiv.org/content/10.1101/2021.03.19.436212v2">Identifying disease-critical cell types and cellular processes by integrating single-cell RNA sequencing and human genetics</a>)。<br>作者开发了sc-linker，一个框架将单细胞数据、表型图谱(epigenomic maps)和GWAS数据集成用以识别潜在的细胞类型和细胞过程(cell process)。该框架再次捕捉了已知的生物学规律并突显重要的细胞-疾病相关性，包括重度抑郁症中的 GABA 能神经元、溃疡性结肠炎中依赖疾病状态的 M 细胞程序，以及多发性硬化症中疾病特异性的补体级联过程。在自身免疫疾病中，健康状态和疾病依赖的免疫细胞类型程序均表现出相关性，而只有疾病依赖的上皮细胞程序显著，这表明它们更多参与疾病的反应过程而非起始阶段。我们的框架提供了一种强有力的方法，用于识别遗传变异影响疾病的细胞类型及其细胞过程。</li>
</ul>
<h1 id="结论"><a href="#结论" class="headerlink" title="结论"></a>结论</h1><h2 id="Overview"><a href="#Overview" class="headerlink" title="Overview"></a>Overview</h2><p>作者构建了一个框架来将从单细胞中获得基因过程(gene programs)与疾病和复杂的特征相关联。首先，作者使用scRNA-seq构建了基因过程，该基因过程定义未一个具有连续值的基因集合，该基因集合特征化了: (1) 个体的细胞类型 (2) 疾病的依赖性(健康人与患者的同种细胞的比较) (3) 细胞过程(cellular processes)。基因过程中的连续值为0-1之间的数值，但并不表示概率。<br>然后，作者将这些基因过程中潜在的基因和调控它的SNP相关联，关联方法依赖于结合两种组织特异性的增强子基因关联策略：1. Roadmap增强子，2. Activity-by-Contact(ABC)模型。<br>最后，作者通过使用S-LDSC，在考虑基线LD模型提供的一系列编码区、保守曲、调控区及连锁不平衡相关注释的条件下，评估所得SNP注释对于疾病信息的贡献。<br>总之，作者的这些操作再次捕捉了已知的生物学规律并突显重要的细胞-疾病相关性。对于这个流程看的不是很明白，后续结合具体数学公式与代码分析。</p>
<h2 id="评价标准-benchmark"><a href="#评价标准-benchmark" class="headerlink" title="评价标准(benchmark)"></a>评价标准(benchmark)</h2><p>为了证明自己的模型优于别人的模型，作者自己做了一个benchmark。作者分析了5种学习的特征，这5个特征对应于特定的免疫细胞类型。作者构建6种免疫细胞类型程序（来自4个数据集：2个PBMC，1个骨髓，1个脐带血），作者从中识别出：红系细胞的富集和红细胞数量有关、巨核细胞和血小板数量有关，单核细胞和单核细胞有关，B、T细胞和淋巴细胞比例有关。这些富集是一种”预期中的富集“，因为这是符合既往的研究和生物学相关性的。<br>同时还定义了敏感性与特异性的index，用以量化现有富集方法和研发出的富集方法的表现效果。</p>
<h1 id="Method"><a href="#Method" class="headerlink" title="Method"></a>Method</h1><p>方法上都是常规思路，但是该论文中对于scRNA的处理策略可以作为后续进行scRNA处理流程的参考</p>
<h2 id="scRNA-seq"><a href="#scRNA-seq" class="headerlink" title="scRNA-seq"></a>scRNA-seq</h2><p>正常处理，然后把基因表达矩阵均转化为$\log_{2}(\text{TP10K}+1)$</p>
<h1 id="降维-去批次-聚类-标注"><a href="#降维-去批次-聚类-标注" class="headerlink" title="降维&#x2F;去批次&#x2F;聚类&#x2F;标注"></a>降维&#x2F;去批次&#x2F;聚类&#x2F;标注</h1><p>作者首先使用Scanpy’s <em>highly_variable_genes</em>函数识别出所有数据集中表达差异最大的2000个基因，然后使用PCA对着2000个基因提取40个主成分，使用Harmony包进行去批次操作，最后再用KNN对于前40个主成分进行聚类分析，聚类为10类。然后对于聚类结果采用Leiden图聚类方法进行社区检测(Community Dectection)，细粒度(resolution)为1。对于每个数据集，都采用UMAP进行可视化操作，如果之前的标注可以获得，那就可以作为参考对每个细胞进行标注，反之则获取每个细胞的独特表达印迹(signatures)与基因markers来进行标注。</p>
<h2 id="Cell-type-gene-programs"><a href="#Cell-type-gene-programs" class="headerlink" title="Cell type gene programs"></a>Cell type gene programs</h2><p>作者对于每个细胞类别进行一次非参数Wilcoxon值和检验比较该细胞类型与其他细胞来获得每个基因在该类细胞中的显著性。然后使用之前提出的策略将p值化为$X&#x3D;-2\log(p)$，这个$X$服从卡方分布，然后再用最大最小归一化方法将X归一化到0-1之间。从健康人哪里获得的细胞类型基因程序被称为健康细胞类型程序，反之则成为疾病细胞类型程序。</p>
<h2 id="Disease-dependent-gene-programs"><a href="#Disease-dependent-gene-programs" class="headerlink" title="Disease-dependent gene programs"></a>Disease-dependent gene programs</h2><p>作者进行健康组织与异常相同组织的非参数检验，然后采用相同的方法转化为0-1之间。作者同时还在饭呢西COVID-19数据集时构建了一个病毒进展程序（比较未感染和已感染的细胞类型）。作者观察到健康细胞类型基因程序与疾病依赖基因程序之间相关性弱。<br><em>Comment: 博主认为，这里的programs其实质上是采用一种特殊的方式将细胞进行一种独特的编码，显著性较高的部分标为1，而不高的部分标为0，这样就形成了一种专属于每种细胞的独特 one-hot encoding。这种独热编码描述了每个细胞的特征，且采用的两种方式的编码具有较弱的关联性，也体现出了：两种不同的编码所描述的细胞特征为不同维度，即正交的</em></p>
<h2 id="Cellular-process-gene-programs"><a href="#Cellular-process-gene-programs" class="headerlink" title="Cellular process gene programs"></a>Cellular process gene programs</h2><p>使用从NMF（非负矩阵分解，non-negative matrix factorization）中获取的潜在因子。作者定义了一种细胞过程程序（cellular process program），该过程基于在不同细胞之间表达量和在每个因（Latent factors）中的贡献性呈现高相关性的基因。这个相关性然后被归一化到0-1（负相关被分配为0），然后作者对于每个因子采用这个因子中“驱动力最强”（most driving）的基因所富集出的通路进行标注。并根据该通路是只与一个细胞种类程序相关还是与多个细胞种类程序相关分为“内细胞类型（intra-cell type）”和”间细胞类型（inter-cell type）”因子。</p>
<ul>
<li>Comment: 这个NMF主要用于图像识别、文本处理，主要思路在于将一个大的矩阵分解为两个非负矩阵的乘积，该方法将一个大的矩阵中提取出重要信息。该思路也在深度学习中有所应用，LoRA作为大模型微调的低廉且快速的策略，其全名便是”Low-Rank Adaptation“。LoRA中，作者认为参数更新中存在一个“内在秩”，也就是说，我们更新的参数，可以采用两个低秩的向量相乘所得，于是我们仅仅需要训练这两个向量，而不需要训练巨大的权重矩阵。<br>我借用Chatgpt的解释：这两种方法都假设“变化在低维子空间里“。NMF：高维观测其实由少数 latent factors 线性组合而成；LoRA：微调任务只需要在权重空间里走一个低维方向。可将其类比为一种 ”基 \times 系数“的类比，也就是说我们将一个庞大的空间分为多个小的基向量，然后与系数相乘来描述原空间。这就保证了我们的预测&#x2F;解释不会在整个参数空间内随意优化，即仅在rank-r子空间内学习这也就是为什么LoRA不易过拟合、NMF可解释性强。<br>再进一步考虑，这种思路其实是类似于谱分析思路，将高维度数据转移到一个低维度&#x2F;低频率的空间进行考虑.</li>
<li></li>
</ul>
<p>这里作者获得细胞过程基因程序采用以下优化函数进行：</p>
<p>$$<br>\operatorname{argmin}\frac{1}{2}\left|X_{n, m}- \sum_{p} W_{n, p} \times H_{p, m}<br>\right|_{F}^{2}+ \frac{1-\alpha}{2}\left|W_{n, p}\right|+ \frac{1-\alpha}{2}\left|H_{p, m}\right|+ \alpha\left|\operatorname{vec}(W_{n, p})\right|_{1}+ \alpha\left|\operatorname{vec}(H_{p, m})\right|_{1}</p>
<p>$$</p>
<p>上图中$X_{n,m}$代表着基因表达矩阵中的元素：基因m在样本n中的表达情况，$W_{n,p}$代表了样本n中的潜在因子p，$H_{p,m}$代表因子p在基因m中的权重。这里第一项表述采用了非矩阵的表示，NMF如果采用矩阵则可以直接如下表示：</p>
<p>$$<br>\operatorname{argmin}\left|X - WH\right|^2 _{F}<br>$$<br>其中 $X\in \mathbb{R}^{N\times M}, W\in \mathbb{R}^{N\times P},H\in \mathbb{R}^{P\times M}$ 最外层表示2范数，也被称为Frobenius范数(F范数)，也就是平方和的平方根。后续其还有多项，我们可以认为是某种正则项，保证优化得到的矩阵不会过拟合。<br>*<br>Comments: 这里的数学公式后面几项是错误的，同时argmin函数的书写不全，可能导致误解，因为W_{n,p}等应为标量(scalar)，而标量不能进行vec操作，取范数操作也不严谨。正确书写形式应该如下：<br>*</p>
<p>$$<br>\begin{aligned}<br>\min_{W, H} &amp; \frac{1}{2}|X-W H|_{F}^{2} \ \<br>&amp; +(1-\alpha) \frac{1}{2}|W|_{F}^{2}+(1-\alpha) \frac{1}{2}|H|_{F}^{2} \ \<br>&amp; +\alpha|\operatorname{vec}(W)|_{1}+\alpha | \operatorname{vec}(H)|_{1}<br>\end{aligned}<br>$$</p>
<p>如上式子展开便可以发现，这个优化函数其实是类似于LASSO + 岭回归的正则化方法（对于系数分别进行1阶和2阶范数的惩罚，目前类似于这样的正则化方法被称为Elastic-Net）。<br>作者通过如下方法定义一个细胞过程基因程序：识别基因中 在每个细胞表达量和factors(也就是W中的一列)呈现强相关。潜在因子中，如果相关性大于0.8（潜在因子之间的，也就是说W中如果出现某2个列相关性大于0.8，那就只考虑其中之一），那么只考虑一个潜在因子。我们采用相关性最强的（这里的相关性是 表达水平和factors之间的相关性）且factors所对应的H中权重更高的基因所富集得出的通路来对于每个细胞过程程序进行标注。并将其标注为”intra-cell type”和”inter-cell type”，前者代表这个细胞过程基因程序只和单个细胞类型程序强相关，后者代表多个细胞类型程序都与之有关。</p>
<h2 id="Cellular-process-gene-programs-constructed-from-healthy-and-disease-tissues"><a href="#Cellular-process-gene-programs-constructed-from-healthy-and-disease-tissues" class="headerlink" title="Cellular process gene programs constructed from healthy and disease tissues"></a>Cellular process gene programs constructed from healthy and disease tissues</h2><p>作者这部分则是将目标放在提取每个数据集中健康人以及患疾病人中的共性与特性。它提出了一种NMF方法方法：<br>$$<br>H_{P\times N_{1}}&#x3D;[L_{P\times K_{C}}^{CH}, L_{P\times K_{H}}^{UH}]F^H_{(K_{C}+K_{H})\times N_{1}}, L^{CH},L^{UH},F^H_{0}&gt;0<br>$$<br>$$<br>H_{P\times N_{2}}&#x3D;[L_{P\times K_{C}}^{CD}, L_{P\times K_{D}}^{UD}]F^D_{(K_{C}+K_{D})\times N_{2}}, L^{CD},L^{UD},F^D_{0}&gt;0<br>$$<br>上文的公式是博主进行了一些“修正”得出的结果，原作者的公式有明显错误。$H_{P\times N_{1}}$代表健康个体中特定组织观察到的基因表达矩阵；$D_{P\times N_{2}}$为相同组织在患病个体中的基因表达矩阵；P代表基因的数目(非常坏的标注习惯，上面不都用过P了，为什么不能用N_Gene_H，这种无歧义的变量名)。这里原作者考虑是将H分解为：$L^{CH}$与$L^{UH}$其中C代表共性、U代表个性，H代表健康，D代表疾病，所以解释就是：$L^{CH}$代表健康人中与患病者的共性部分，$L^{UH}$代表健康人中个性的部分；$L^{CD}$代表患病者中与健康人的共性部分，$L^{UD}$代表患病者中的个性部分。所以作者希望共性部分能尽量接近，这也在后面的优化函数中有所体现。<br>$$<br>\begin{array}{l}\quad \underset{L^{H}, L^{D}, F^{H}, F^{D}}{\operatorname{argmin}} \frac{1}{2}\left|H-L^{H} F^{H}\right|_{F}^{2}+\frac{1}{2}\left|D-L^{D} F^{D}\right|_{F}^{2}+\frac{\mu}{2}\left(\left|L^{H}\right|_{F}^{2}+\left|L^{D}\right|_{F}^{2}\right) \ \ +\frac{\gamma}{2}\left(\left|{L}^{C H}-L^{C D}\right|_{F}^{2}\right)\end{array}<br>$$<br>其中，$L^H &#x3D; [L^{CH}_{P\times K_{C}}, L^{UH}_{P\times K_{H}}]$，$L^D&#x3D;[L^{CD}_{P\times K_{C}}, L^{UD}_{P\times K_{D}}]$，$\gamma$为可调参数，用以控制共性部分的相似性；$\mu$依旧控制正则化部分。<br>对于以上所有待优化参数进行求导得到倒数:<br>$$<br>\begin{array}{c}<br>\nabla _{L^{H}}&#x3D;-H F^{H^{T}}+L^{H} F^{H} F^{H^{T}}+\mu L^{H}-\gamma\left[L^{CH} - L^{C D} ;0\right] \ \<br> \nabla _{L^{D}}&#x3D;-D F^{D^{T}}+L^{D} F^{D} F^{D^{T}}+\mu L^{D}-\gamma\left[L^{C D} -L^{C H}; 0\right] \ \<br> \nabla_{F^{H}}&#x3D;-L^{H^{T}} H+L^{H^{T}} L^{H} F^{H} \  \<br>\nabla_{F^{D}}&#x3D;-L^{D^{T}} D+L^{D^{T}} L^{D} F^{D}<br>\end{array}<br>$$<br><em>Comments: 作者后面使用了NMF自带的参数更新策略，我在验证后发现，最终的优化模型大致一致，但是模型中对于前项求导错误</em><br>作者应采用以下公式进行参数更新，而非原公式，直到100个周期或者误差小于一个特定值。<br>$$<br>\begin{align}<br>L_{i j}^{H} &amp;\leftarrow L_{i j}^{H} \frac{\left(H F^{H^{T}}\right)_{i j}}{\left(L^{H} F^{H} F^{H^{T}}+\mu L^{H} + \gamma\left[L^{C H} - L^{C D}; 0\right]\right)_{i j}} \\<br>F_{i j}^{H} &amp;\leftarrow F_{i j}^{H} \frac{\left(L^{H^{T}} H\right)_{i j}}{\left(L^{H^{T}} L^{H} F^{H}\right)_{i j}}<br>\end{align}<br>$$</p>
<h2 id="Enhancer-gene-linking-stategies"><a href="#Enhancer-gene-linking-stategies" class="headerlink" title="Enhancer-gene linking stategies"></a>Enhancer-gene linking stategies</h2><p>作者使用一种增强子-基因连接策略，将0或多个基因分配给在1000 Genomes Project 欧洲人群参考中次等位基因计数大于5的每一个SNP。这里，我们主要考虑了Roadmap + ABC的策略，这两种方法一方面广泛可用，另一方面是优于其他方法。对于与相关组织对应的任何生物样本（细胞类型或组织），我们均采用组织特异性的 Roadmap 和 ABC 增强子–基因链接策略来构建相应的基因程序（gene programs）。基于对于免疫细胞的类型的分析，在单细胞测序中的87%的基因可以被观察到有增强子-基因连接。同时，我们也考虑了非组织特异的Roadmap和ABC策略，除了这个，作者还考虑标准的100kb窗宽的策略<br>这里的增强子-基因连接策略用于分析SNP与特定基因的相关性，是连接GWAS与scRNA的重要桥梁。</p>
<h2 id="Genomic-annotations-and-the-baseline-LD-models"><a href="#Genomic-annotations-and-the-baseline-LD-models" class="headerlink" title="Genomic annotations and the baseline-LD models"></a>Genomic annotations and the baseline-LD models</h2><p>作者将会给每个SNP赋予一个特定的值，这个过程便被称为标注，这个标注如果对于了某种功能，则被称为功能标注。基线-LD模型包括了86种标注。</p>
<h2 id="Stratified-LD-score-regression"><a href="#Stratified-LD-score-regression" class="headerlink" title="Stratified LD score regression"></a>Stratified LD score regression</h2><p>分层连锁不平衡分数回归（S-LDSC）评估了一个基因标注对于疾病或者其他表征的贡献性。<br>作者使用了两个度量来评估标注c的信息量，分别为E-score和标准化效应量$\tau^*$</p>
<h2 id="MAGMA-基因富集"><a href="#MAGMA-基因富集" class="headerlink" title="MAGMA 基因富集"></a>MAGMA 基因富集</h2><p>MAGMA评估了疾病中基因富集的情况</p>
<h2 id="评估敏感性与特异性"><a href="#评估敏感性与特异性" class="headerlink" title="评估敏感性与特异性"></a>评估敏感性与特异性</h2><p>作者定义了一个敏感性与特异性的benchmark。1. sc-linker与MAGMA基因富集分析的比较；2. 不同版本中采用不同方法来定义细胞类型程序和SNP-to-gene连接策略的sc-linkers之间的比较。<br>为了和MAGMA比较，作者比较了1. 对于期待通路的sc-linker中的富集评分和MAGMA中的相关性之间的差值(敏感性，也就是该找的都该找到)；2. 对于其他通路在基因集等级上的sc-linker中的富集评分和MAGMA中的相关性之间的差值（特异性，也就是不该找的有没有没找到）。作者承认有未被观测到的未知富集，所以作者也会考虑在寻找目标富集的敏感性作为代替。</p>
<h2 id="Identifying-genes-driving-heritability-enrichment"><a href="#Identifying-genes-driving-heritability-enrichment" class="headerlink" title="Identifying genes driving heritability enrichment"></a>Identifying genes driving heritability enrichment</h2><p>对于每个基因程序，我们首先从完整的基因列表中进行筛选，仅考虑那些在该基因程序中<strong>具有超过 80% 成员概率等级（probability grade of membership）</strong> 的基因。接着用MAGMA对于留下的基因进行排序，并考虑最高的50个基因来进行下一步的下游分析。而baseline则是取出前200个基因进行针对细胞类型的疾病通路富集。</p>
<hr>

<p>Cover image icon by <a target="_blank" rel="noopener" href="https://www.flaticon.com/free-icons/inference" title="inference icons">Dewi Sari</a> from <a target="_blank" rel="noopener" href="https://www.flaticon.com/free-icons/inference" title="inference icons">Flaticon</a></p>
</article><div class="post-copyright"><div class="post-copyright__author"><span class="post-copyright-meta"><i class="fas fa-circle-user fa-fw"></i>文章作者: </span><span class="post-copyright-info"><a href="http://evanescence0515.github.io">Henry</a></span></div><div class="post-copyright__type"><span class="post-copyright-meta"><i class="fas fa-square-arrow-up-right fa-fw"></i>文章链接: </span><span class="post-copyright-info"><a href="http://evanescence0515.github.io/2026/01/12/GSSG/">http://evanescence0515.github.io/2026/01/12/GSSG/</a></span></div><div class="post-copyright__notice"><span class="post-copyright-meta"><i class="fas fa-circle-exclamation fa-fw"></i>版权声明: </span><span class="post-copyright-info">本博客所有文章除特别声明外，均采用 <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" target="_blank">CC BY-NC-SA 4.0</a> 许可协议。转载请注明来源 <a href="http://evanescence0515.github.io" target="_blank">Evanescence's Blog</a>！</span></div></div><div class="tag_share"><div class="post-share"><div class="social-share" data-image="/img/epidemic_pic.png" data-sites="facebook,twitter,wechat,weibo,qq"></div><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/butterfly-extsrc/sharejs/dist/css/share.min.css" media="print" onload="this.media='all'"><script src="https://cdn.jsdelivr.net/npm/butterfly-extsrc/sharejs/dist/js/social-share.min.js" defer></script></div></div><nav class="pagination-post" id="pagination"><a class="pagination-related full-width" href="/2025/12/15/MSHTrans/" title="MSHTrans文献速递(KDD, 2025)"><img class="cover" src="/img/deeplearning_pic.png" onerror="onerror=null;src='/img/404.jpg'" alt="cover of next post"><div class="info text-right"><div class="info-1"><div class="info-item-1">下一篇</div><div class="info-item-2">MSHTrans文献速递(KDD, 2025)</div></div><div class="info-2"><div class="info-item-1">MSHTrans: 带有时间序列分解的多尺度超图Transformer实现时间异常检测。 Abstract目前已有的异常检测方法在理解重构模型的机理过程中存在2个主要挑战：1. 对于时间标签的长期和短期依赖性的挖掘不足；2. 同时挖掘长期与短期特征的框架缺失。为了解决这个问题，作者提出了MSHTrans，其充分使用了超图建模多阶时间依赖关系方面的能力。作者的模型使用了多尺度下采样来获取互为补充的细粒度和粗粒度表示，并与可训练的超图神经网络结合，该网络可以自适应地学习时间标签之间的关联性。这个框架进一步结合了时间序列分解来从多粒度特征中系统地提取周期性与趋势性成分。 Introduction时间序列异常检测是真实世界领域中的一个基础工作，尤其是在工业、医疗和金融领域，该领域主要有3大问题：1. 大规模数据；2. 异质性；3. 多模态特征。为了解决这3大问题，目前已有的模型可以被简单分为2类：基于预测的方法：该方法采用历史数据预测未来数据并根据预测误差来定量评估异常，但是这个方法对于时间特征快速改变的情形并不适用；基于重构的方法在近年获得重视，其通过从学习到的潜在表示重构数据并与真实...</div></div></div></a></nav></div><div class="aside-content" id="aside-content"><div class="card-widget card-info text-center"><div class="avatar-img"><img src="/img/my_icon.jpg" onerror="this.onerror=null;this.src='/img/friend_404.gif'" alt="avatar"/></div><div class="author-info-name">Henry</div><div class="author-info-description">至此鲜花赠自己，纵马踏花向自由</div><div class="site-data"><a href="/archives/"><div class="headline">文章</div><div class="length-num">35</div></a><a href="/tags/"><div class="headline">标签</div><div class="length-num">9</div></a><a href="/categories/"><div class="headline">分类</div><div class="length-num">3</div></a></div><a id="card-info-btn" target="_blank" rel="noopener" href="https://github.com/evanescence0515"><i class="fab fa-github"></i><span>Follow Me</span></a><div class="card-info-social-icons"><a class="social-icon" href="https://github.com/evanescence0515" target="_blank" title="Github"><i class="fab fa-github" style="color: #24292e;"></i></a><a class="social-icon" href="mailto:evanescence0515@gmail.com" target="_blank" title="Email"><i class="fas fa-envelope" style="color: #4a7dbe;"></i></a></div></div><div class="card-widget card-announcement"><div class="item-headline"><i class="fas fa-bullhorn fa-shake"></i><span>公告</span></div><div class="announcement_content">你好呀，这里是Evanescence的博客，  我会经常分享一些学习统计学、生物信息学、机器学习和计算机技术的心得和笔记。</div></div><div class="sticky_layout"><div class="card-widget" id="card-toc"><div class="item-headline"><i class="fas fa-stream"></i><span>目录</span><span class="toc-percentage"></span></div><div class="toc-content"><ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link" href="#%E7%AE%80%E4%BB%8B"><span class="toc-number">1.</span> <span class="toc-text">简介</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#%E7%BB%93%E8%AE%BA"><span class="toc-number">2.</span> <span class="toc-text">结论</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#Overview"><span class="toc-number">2.1.</span> <span class="toc-text">Overview</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E8%AF%84%E4%BB%B7%E6%A0%87%E5%87%86-benchmark"><span class="toc-number">2.2.</span> <span class="toc-text">评价标准(benchmark)</span></a></li></ol></li><li class="toc-item toc-level-1"><a class="toc-link" href="#Method"><span class="toc-number">3.</span> <span class="toc-text">Method</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#scRNA-seq"><span class="toc-number">3.1.</span> <span class="toc-text">scRNA-seq</span></a></li></ol></li><li class="toc-item toc-level-1"><a class="toc-link" href="#%E9%99%8D%E7%BB%B4-%E5%8E%BB%E6%89%B9%E6%AC%A1-%E8%81%9A%E7%B1%BB-%E6%A0%87%E6%B3%A8"><span class="toc-number">4.</span> <span class="toc-text">降维&#x2F;去批次&#x2F;聚类&#x2F;标注</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#Cell-type-gene-programs"><span class="toc-number">4.1.</span> <span class="toc-text">Cell type gene programs</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#Disease-dependent-gene-programs"><span class="toc-number">4.2.</span> <span class="toc-text">Disease-dependent gene programs</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#Cellular-process-gene-programs"><span class="toc-number">4.3.</span> <span class="toc-text">Cellular process gene programs</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#Cellular-process-gene-programs-constructed-from-healthy-and-disease-tissues"><span class="toc-number">4.4.</span> <span class="toc-text">Cellular process gene programs constructed from healthy and disease tissues</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#Enhancer-gene-linking-stategies"><span class="toc-number">4.5.</span> <span class="toc-text">Enhancer-gene linking stategies</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#Genomic-annotations-and-the-baseline-LD-models"><span class="toc-number">4.6.</span> <span class="toc-text">Genomic annotations and the baseline-LD models</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#Stratified-LD-score-regression"><span class="toc-number">4.7.</span> <span class="toc-text">Stratified LD score regression</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#MAGMA-%E5%9F%BA%E5%9B%A0%E5%AF%8C%E9%9B%86"><span class="toc-number">4.8.</span> <span class="toc-text">MAGMA 基因富集</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E8%AF%84%E4%BC%B0%E6%95%8F%E6%84%9F%E6%80%A7%E4%B8%8E%E7%89%B9%E5%BC%82%E6%80%A7"><span class="toc-number">4.9.</span> <span class="toc-text">评估敏感性与特异性</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#Identifying-genes-driving-heritability-enrichment"><span class="toc-number">4.10.</span> <span class="toc-text">Identifying genes driving heritability enrichment</span></a></li></ol></li></ol></div></div><div class="card-widget card-recent-post"><div class="item-headline"><i class="fas fa-history"></i><span>最新文章</span></div><div class="aside-list"><div class="aside-list-item"><a class="thumbnail" href="/2026/01/12/GSSG/" title="GSSG"><img src="/img/epidemic_pic.png" onerror="this.onerror=null;this.src='/img/404.jpg'" alt="GSSG"/></a><div class="content"><a class="title" href="/2026/01/12/GSSG/" title="GSSG">GSSG</a><time datetime="2026-01-12T09:33:57.000Z" title="发表于 2026-01-12 17:33:57">2026-01-12</time></div></div><div class="aside-list-item"><a class="thumbnail" href="/2025/12/15/MSHTrans/" title="MSHTrans文献速递(KDD, 2025)"><img src="/img/deeplearning_pic.png" onerror="this.onerror=null;this.src='/img/404.jpg'" alt="MSHTrans文献速递(KDD, 2025)"/></a><div class="content"><a class="title" href="/2025/12/15/MSHTrans/" title="MSHTrans文献速递(KDD, 2025)">MSHTrans文献速递(KDD, 2025)</a><time datetime="2025-12-15T10:53:53.000Z" title="发表于 2025-12-15 18:53:53">2025-12-15</time></div></div><div class="aside-list-item"><a class="thumbnail" href="/2025/12/15/MSHTrans%E4%BB%A3%E7%A0%81%E9%98%85%E8%AF%BB/" title="MSHTrans代码阅读与模块解读(KDD, 2025)"><img src="/img/deeplearning_pic.png" onerror="this.onerror=null;this.src='/img/404.jpg'" alt="MSHTrans代码阅读与模块解读(KDD, 2025)"/></a><div class="content"><a class="title" href="/2025/12/15/MSHTrans%E4%BB%A3%E7%A0%81%E9%98%85%E8%AF%BB/" title="MSHTrans代码阅读与模块解读(KDD, 2025)">MSHTrans代码阅读与模块解读(KDD, 2025)</a><time datetime="2025-12-15T10:51:08.000Z" title="发表于 2025-12-15 18:51:08">2025-12-15</time></div></div><div class="aside-list-item"><a class="thumbnail" href="/2025/12/15/GCN-lesson2/" title="torch geometric学习笔记2(MessagePassing)"><img src="/img/deeplearning_pic.png" onerror="this.onerror=null;this.src='/img/404.jpg'" alt="torch geometric学习笔记2(MessagePassing)"/></a><div class="content"><a class="title" href="/2025/12/15/GCN-lesson2/" title="torch geometric学习笔记2(MessagePassing)">torch geometric学习笔记2(MessagePassing)</a><time datetime="2025-12-15T10:37:25.000Z" title="发表于 2025-12-15 18:37:25">2025-12-15</time></div></div><div class="aside-list-item"><a class="thumbnail" href="/2025/12/15/PLR/" title="PLR嵌入策略论文速递(NeurIPS, 2022)"><img src="/img/deeplearning_pic.png" onerror="this.onerror=null;this.src='/img/404.jpg'" alt="PLR嵌入策略论文速递(NeurIPS, 2022)"/></a><div class="content"><a class="title" href="/2025/12/15/PLR/" title="PLR嵌入策略论文速递(NeurIPS, 2022)">PLR嵌入策略论文速递(NeurIPS, 2022)</a><time datetime="2025-12-15T10:26:50.000Z" title="发表于 2025-12-15 18:26:50">2025-12-15</time></div></div></div></div></div></div></main><footer id="footer"><div class="footer-other"><div class="footer-copyright"><span class="copyright">&copy;2019 - 2026 By Henry</span></div></div></footer></div><div id="rightside"><div id="rightside-config-hide"><button id="readmode" type="button" title="阅读模式"><i class="fas fa-book-open"></i></button><button id="darkmode" type="button" title="日间和夜间模式切换"><i class="fas fa-adjust"></i></button><button id="hide-aside-btn" type="button" title="单栏和双栏切换"><i class="fas fa-arrows-alt-h"></i></button></div><div id="rightside-config-show"><button id="rightside-config" type="button" title="设置"><i class="fas fa-cog fa-spin"></i></button><button class="close" id="mobile-toc-button" type="button" title="目录"><i class="fas fa-list-ul"></i></button><button id="go-up" type="button" title="回到顶部"><span class="scroll-percent"></span><i class="fas fa-arrow-up"></i></button></div></div><div><script src="/js/utils.js"></script><script src="/js/main.js"></script><div class="js-pjax"><script>(() => {
  const loadMathjax = () => {
    if (!window.MathJax) {
      window.MathJax = {
        tex: {
          inlineMath: [['$', '$'], ['\\(', '\\)']],
          tags: 'none',
        },
        chtml: {
          scale: 1.1
        },
        options: {
          enableMenu: true,
          renderActions: {
            findScript: [10, doc => {
              for (const node of document.querySelectorAll('script[type^="math/tex"]')) {
                const display = !!node.type.match(/; *mode=display/)
                const math = new doc.options.MathItem(node.textContent, doc.inputJax[0], display)
                const text = document.createTextNode('')
                node.parentNode.replaceChild(text, node)
                math.start = {node: text, delim: '', n: 0}
                math.end = {node: text, delim: '', n: 0}
                doc.math.push(math)
              }
            }, '']
          }
        }
      }

      const script = document.createElement('script')
      script.src = 'https://cdn.jsdelivr.net/npm/mathjax/es5/tex-mml-chtml.min.js'
      script.id = 'MathJax-script'
      script.async = true
      document.head.appendChild(script)
    } else {
      MathJax.startup.document.state(0)
      MathJax.texReset()
      MathJax.typesetPromise()
    }
  }

  btf.addGlobalFn('encrypt', loadMathjax, 'mathjax')
  window.pjax ? loadMathjax() : window.addEventListener('load', loadMathjax)
})()</script></div><script async data-pjax src="//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script><div id="local-search"><div class="search-dialog"><nav class="search-nav"><span class="search-dialog-title">搜索</span><span id="loading-status"></span><button class="search-close-button"><i class="fas fa-times"></i></button></nav><div class="text-center" id="loading-database"><i class="fas fa-spinner fa-pulse"></i><span>  数据加载中</span></div><div class="search-wrap"><div id="local-search-input"><div class="local-search-box"><input class="local-search-box--input" placeholder="搜索内容..." type="text"/></div></div><hr/><div id="local-search-results"></div><div id="local-search-stats-wrap"></div></div></div><div id="search-mask"></div><script src="/js/search/local-search.js"></script></div></div></body></html>